server/app.py

```diff
@@ -74,10 +75,27 @@ def remove_player_from_lobby(data):
     room = data['room']
     player_name = data['playerName']
     # Remove the player from the room
-    ROOMS[room].pop(player_name, None)
-    SOCKETIO.emit('removePlayerFromLobby', {'activePlayers': list(ROOMS[room].keys()), 'room': room}, room=room) 
+    ROOMS[room].pop(player_name)
+    socketio.emit('removePlayerFromLobby', {'activePlayers': list(ROOMS[room].keys()), 'room': room}, room=room)
+
     leave_room(room)
 
+@socketio.on('logout')
+def logout_player(data):
+    '''User logouts out of the entire game'''
+    player_name = data['playerName']
+
+    for i in rooms(): #For each room the user is in
+        if i in ROOMS and player_name in ROOMS[i]: #If the user is in the room and that room exists in list of ROOMS
+            ROOMS[i].pop(player_name) #Get rid of the user in that room
+            #Emit to all the users and get rid of that user if they in the room
+            socketio.emit('updatePlayerStats', {'playerStats': ROOMS[i]})
+            socketio.emit('assignPlayerToLobby', {'activePlayers': list(ROOMS[i].keys()), 'room': i})
+
+            leave_room(i) #Leave the room
```

Comment:
I think we should take advantage of the fact that the client knows what room it is in. We can just pass the room code with `data` here instead of iterating through the entire dictionary to find which room the player is in.

Reply:
Makes sense, no need to overwork the server when the client knows which room it's in.

Reply:
But see how I am emitting from login screen, the login sereen doesn't have access to the rooms variable from home screen, if we need that we need to pass the rooms back to it's parent or call the function in homescreen or something

Reply:
I think it makes sense to move the logout button into HomeScreen.js anyway, where it can then emit `room` to the server.